packages/types/src/request-logic-types.ts

```diff
@@ -140,6 +140,33 @@ export interface IRequest {
   nonce?: number;
 }
 
+/** Pending request in request logic */
+export interface IPendingRequest {
```

Comment:
Why every attributes are optional ? We always have a request id isn't it ?

Reply:
You can use `Partial<IPendingRequest>` to make all fields optional

Reply:
probably something in the style of `type IPendingRequest = Partial<IRequest>` could work?